helper.php

```diff
@@ -70,16 +70,34 @@ public function removePage($page)
     }
 
     /**
-     * Update last modified date of page
+     * Update last modified date of page if content has changed
      *
      * @param string $page Page ID
      * @param int $lastmod timestamp of last non-minor change
      */
-    public function storePageDate($page, $lastmod)
+    public function storePageDate($page, $lastmod, $newContent)
     {
         $sqlite = $this->getDB();
         if (!$sqlite) return;
 
+        $identical = true;
+        $changelog = new \dokuwiki\ChangeLog\PageChangeLog($page);
+
+        $revs = $changelog->getRevisions(1, 20);
+
+        foreach ($revs as $rev) {
+            $info = $changelog->getRevisionInfo($rev);
+            if ($info['type'] !== DOKU_CHANGE_TYPE_MINOR_EDIT) {
+                // compare content
+                $oldContent = str_replace(NL, '', io_readFile(wikiFN($page, $rev)));
+                $newContent = str_replace(NL, '', $newContent);
+                if ($oldContent !== $newContent) $identical = false;
+                break;
+            }
+        }
+
+        if ($identical) return;
```

Comment:
if the last 20 revisions were minor, or no previous revisions exist at all, $identical will be true but maybe shouldn't.